sympy/holonomic/holonomic.py

```diff
@@ -2186,11 +2186,11 @@ def from_hyper(func, x0=0, evalf=False):
 
     # generalized hypergeometric differential equation
     r1 = 1
-    for i in range(len(a)):
-        r1 = r1 * (x * Dx + a[i])
+    for ai in a:
+        r1 = r1 * (x * Dx + ai)
```

Comment:
`xdx = x*Dx; r1 = Mul(*[xdx + ai for ai in a])` and similar for `r2` but also multiply by Dx when done.

Reply:
same for the other ones below. As I recall, building up the Add/Mul is a slow way to go; better to build args then do op all at once.